haproxy_exporter.go

```diff
@@ -402,20 +401,29 @@ func (e *Exporter) exportCsvFields(metrics map[int]*prometheus.GaugeVec, csvRow
 			continue
 		}
 
-		var value int64
+		var value float64
 		switch fieldIdx {
 		case statusField:
 			value = parseStatusField(valueStr)
 		default:
 			var err error
-			value, err = strconv.ParseInt(valueStr, 10, 64)
+			value, err = strconv.ParseFloat(valueStr, 64)
 			if err != nil {
 				log.Errorf("Can't parse CSV field value %s: %v", valueStr, err)
 				e.csvParseFailures.Inc()
 				continue
 			}
 		}
-		metric.WithLabelValues(labels...).Set(float64(value))
+		// Handle value translations.
+		switch csvRow[32] {
```

Comment:
This check is unnecessary, fields 58-61 are always timers, in all row types, not just in backend.